### ISSUE TITLE
Set number of visual grid columns with media queries

### ISSUE THREAD

**Issue by jink13** [thoughtbot/neat#328]
Hi ! 
I know it's a stupid question but.. ( My english is quite bad..)
How to change the number of the visual grid's columns when we use media queries ?
Unlike the example page (http://neat.bourbon.io/examples/) in the Media queries section, when i resize my browser window, it doesn't switch to 4 columns. 
I just do not understand why ?
I know we can change the variable $grid-columns, but i would like it to change when i resize the browser.
I'm sure there is an easy way out but i don't manage to find it with the documentation and the comments.
Thanks !

**Comment by mehlah** [comment]
That comes from the `$grid-columns` variable of `new-breakpoint` function.
This is how the breakpoint is defined for the docs example: https://github.com/thoughtbot/neat/blob/gh-pages/_sass/_variables.scss#L7

**Comment by jink13** [comment]
Thanks ! But it doesn't work.
I don't know, there is a conflict or something. I read it again and again. Everything is set up. I haven't forget semicolon. Idon't know.. I'me sur it's very easy but.. I can't see where the problem is.
I declare my variable $mobile: new-breakpoint(max-width 500px 4); at the begining of my stylesheet (main.scss), just after @import 'bourbon/bourbon' @import 'neat/neat' .
I added  my media querie to my element. 
.element {
       @include media($mobile) {
			@include span-columns(3);
	}
}

I resize my browser, and the result is ? Nothing.
I don't get it..
![screen-capture](https://cloud.githubusercontent.com/assets/10714016/6902127/2061b2f6-d712-11e4-8cb8-6461af0df043.png)

**Comment by millertchris** [comment]
@mehlah I'm currently experiencing the same issue as @jink13 is experiencing. 

I'll try to give you as much context as possible with a screenshot.

Read through the documentation 10x, plus endless Google searching. I've followed the documentation, verbatim (although I could be making some ridiculous mistakes).

You will notice that I'm setting the new breakpoint after importing neat, if I put it before neat...it does not honor the breakpoint in sections where I'm calling that variable.

Therefor, I've set it after. However, it would make sense to come before Neat as the documentation points out...but it doesn't seem to work.

Thoughts or ideas?

Here is a screenshot:
![screen shot 2015-04-01 at 12 56 13 pm](https://cloud.githubusercontent.com/assets/3119509/6946506/c0c24776-d86e-11e4-9cc5-40c2b9b55b1f.png)

**Comment by jink13** [comment]
@millertchris I found the solution. I'll post a link as soon as i'll get back home.

**Comment by millertchris** [comment]
@jink13 That would be stellar!!! Thanks!

**Comment by jink13** [comment]
This "THE" link : 
https://gist.github.com/andyharris/3995673#gistcomment-592150
It helps me to understand the "trick".
@kaisin explains a solution in his comment (1 nov 2012)
Now it works. Good luck !
![capture](https://cloud.githubusercontent.com/assets/10714016/6951484/f5c349aa-d8bb-11e4-86ad-5415825d8bae.png)

**Comment by tysongach** [comment]
@jink13 @millertchris Yes, per our [readme](https://github.com/thoughtbot/neat#using-neat), if you’re planning to change the default grid settings, you need to do so before `@import`’ing Neat.

Try that and let me know if it resolves your issue.

**Comment by jink13** [comment]
Thank's @tysongach ! 
Yes. I use this technic and It works. I replaced the file _grid-settings.scss to _variables as you can see in my previous capture (in my last comment). But it's the same. 
Unfortunately, i hadn't seen this part of the documentation. Thanks!

**Comment by millertchris** [comment]
@tysongach @jink13 @mehlah Alright, I've finally figured it out, but the documentation is deceptive or lacks in clear direction.

I'm thoroughly enjoying Bourbon, prior to using Bourbon I had been using Susy but find Bourbon to be more enjoyable!

Alright...so, here are my recommendations:

Rather, telling the audience to simply import before Neat (which didn't work for me), instead, let them know to @import "neat-helpers"; then $mobile: new-breakpoint(min-width 480px 4);, and finally @import "neat";

That's the only way I was able to get it to work.

To illustrate this from the top down:

@import "bourbon";
@import "neat-helpers";
$max-width: 1140px;
$mobile: new-breakpoint(max-width 320px 4);
$tablet: new-breakpoint(min-width 480px 8);
$desktop: new-breakpoint(min-width 720px 12);
$visual-grid: true;
$visual-grid-color: #eee;
$visual-grid-index: front;
$visual-grid-opacity: 0.75;
@import "neat";

If you stuff the above "settings" into a .scss file, then it would like something like this:

@import "bourbon";
@import "neat-helpers";
@import "settings";
@import "neat";

In conclusion...yes...put your settings before "neat", however, you will also need to include "neat-helpers" before settings as well. Even though "neat-helpers" is included and imported once again within "neat".

I've yet to really dig into the functions and structure of how it is all setup, I've simply just been working with Bourbon with what documentation is out there, but the documentation is still slightly vague.

With all of that said, Bourbon is GREAT! It's a great framework with tons of potential, I hope to contribute to it one day, in anyway possible.

**Comment by tysongach** [comment]
@millertchris Yes, this is correct. Sorry for the docs not being as clear as they could be…

You need the `neat-helpers` because it houses the custom functions being used to define these settings, e.g. `new-breakpoint`.

**Comment by tysongach** [comment]
Actually, our readme has a line about this: “In your newly created _grid-settings.scss, import neat-helpers if you are planning to use new-breakpoint(), then define your new variables:”

It’s also worth noting that Bitters comes pre-packaged with a `_grid-settings.scss` file: https://github.com/thoughtbot/bitters/blob/master/app/assets/stylesheets/_grid-settings.scss

**Comment by millertchris** [comment]
Sweet! Again, loving it and hope to contribute to it very soon!
